dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/listener/EnrollmentSMSListener.java

```diff
@@ -101,6 +111,14 @@ protected boolean handlesType(Consts.SubmissionType type) {
         return (type == Consts.SubmissionType.ENROLLMENT);
     }
 
+    private Set<TrackedEntityAttributeValue> getAttributeValues(EnrollmentSMSSubmission submission, TrackedEntityInstance entityInstance) {
+        return submission.getValues()
+                .stream()
+                .filter(this::isValidAttribute)
+                .map(v -> createTrackedEntityValue(v, entityInstance))
```

Comment:
It feels weird that you need the TEI when creating these attribute values but then you still need to set them against the TEI later... I wonder what's with that?